src/ui/control/geolocate_control.js

```diff
@@ -259,7 +259,10 @@ class GeolocateControl extends Evented {
     }
 
     _setupUI(supported: boolean) {
-        if (supported === false) return;
+        if (supported === false) {
+            warnOnce('Geolocation not supported. The Mapbox geolocate control will not be rendered.');
```

Comment:
+1 for adding this warning, since the developer has asked for the GeolocateControl, we should at least warn when that hasn't happened.

Could we change the warning to `Geolocation support is not available, the GeolocateControl will not be visible.`?

Reply:
@andrewharvey Absolutely, will update that now.